haystack/components/generators/chat/hugging_face_local.py

```diff
@@ -0,0 +1,366 @@
+            See Hugging Face's [documentation](https://huggingface.co/docs/transformers/en/main_classes/pipelines#transformers.pipeline.task)
+            for more information on the available kwargs.
+            In this dictionary, you can also include `model_kwargs` to specify the kwargs
+            for model initialization:
+            https://huggingface.co/docs/transformers/en/main_classes/model#transformers.PreTrainedModel.from_pretrained
+        :param stop_words: A list of stop words. If any one of the stop words is generated, the generation is stopped.
+            If you provide this parameter, you should not specify the `stopping_criteria` in `generation_kwargs`.
+            For some chat models, the output includes both the new text and the original prompt.
+            In these cases, it's important to make sure your prompt has no stop words.
+        :param streaming_callback: An optional callable for handling streaming responses.
+        """
+        torch_and_transformers_import.check()
+
+        huggingface_pipeline_kwargs = huggingface_pipeline_kwargs or {}
+        generation_kwargs = generation_kwargs or {}
+
+        # check if the huggingface_pipeline_kwargs contain the essential parameters
+        # otherwise, populate them with values from other init parameters
+        huggingface_pipeline_kwargs.setdefault("model", model)
+        huggingface_pipeline_kwargs.setdefault("token", token)
+        if (
+            device is not None
+            and "device" not in huggingface_pipeline_kwargs
+            and "device_map" not in huggingface_pipeline_kwargs
+        ):
+            huggingface_pipeline_kwargs["device"] = device
+
+        # task identification and validation
+        if task is None:
+            if "task" in huggingface_pipeline_kwargs:
+                task = huggingface_pipeline_kwargs["task"]
+            elif isinstance(huggingface_pipeline_kwargs["model"], str):
+                task = model_info(
+                    huggingface_pipeline_kwargs["model"], token=huggingface_pipeline_kwargs["token"]
+                ).pipeline_tag
+
+        if task not in PIPELINE_SUPPORTED_TASKS:
+            raise ValueError(
+                f"Task '{task}' is not supported. " f"The supported tasks are: {', '.join(PIPELINE_SUPPORTED_TASKS)}."
+            )
+        huggingface_pipeline_kwargs["task"] = task
+
+        # if not specified, set return_full_text to False for text-generation
+        # only generated text is returned (excluding prompt)
+        if task == "text-generation":
+            generation_kwargs.setdefault("return_full_text", False)
+
+        if stop_words and "stopping_criteria" in generation_kwargs:
+            raise ValueError(
+                "Found both the `stop_words` init parameter and the `stopping_criteria` key in `generation_kwargs`. "
+                "Please specify only one of them."
+            )
+        generation_kwargs.setdefault("max_new_tokens", 512)
```

Comment:
Adopting a default number of `max_new_tokens` is reasonable.
- if we want to add this default, let's document it in the docstrings
- (let's also open a PR to do the same in the HFLocalGenerator)